keps/sig-auth/0000-20190116-service-account-key-pipereader.md

```diff
@@ -0,0 +1,129 @@
+---
+kep-number: 0
+title: Read service account key files from unix socket
+authors:
+  - "@micahhausler"
+owning-sig: sig-auth
+participating-sigs: []
+reviewers:
+  - "@mikedanese"
+  - "@liggit"
+  - "@tallclair"
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-01-16
+last-updated: 2019-01-16
+status: provisional
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Read Service Account key files from a unix socket
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Preserve existing behavior](#preserve-existing-behavior)
+  * [Updates to API server token generation](#updates-to-api-server-token-generation)
+  * [New API](#new-api)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+
+The Kubernetes API server has always read service account keys from disk as the process starts. As the API server can now verify and issue projected volume tokens, it would be advantageous to read these signing and verifying keys from a unix socket rather than a file.
+
+## Motivation
+
+For operators who want to regularly rotate the signing and verifying key files for projected volume tokens, the API server must be restarted in order to use a new key. To facilitate easy key rotation, this KEP includes an proposal for a grpc API to read keys.
+
+### Goals
+
+- A new API for reading signing and verifiying keys over a unix socket
+- Preserve existing behavior and performance for keys not read over a socket
+
+### Non-Goals
+
+- Reading TLS serving certificates and key from a socket or reloading of the API server with new cert and key
+- Reading any other certificates from a file
+- Supporting out-of-process JWT signing and verifying
```

Comment:
This is very desirable in my opinion. A plugin that supports out of process signing can support downloaded keys (where the plugin driver does the signing) but not the other way around.